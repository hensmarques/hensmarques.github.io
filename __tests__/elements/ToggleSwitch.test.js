import React from 'react';
import ToggleSwitch from '../../src/elements/ToggleSwitch';

describe('<ToggleSwitch/>', () => {
  const mockProps = {
    value: true,
    onChange: jest.fn()
  };

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ToggleSwitch {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('fires onChange when the switch is toggled', () => {
    const wrapper = shallow(<ToggleSwitch {...mockProps} />);
    expect(wrapper.find('ReactSwitch').props().checked).toBe(true);
    wrapper.find('ReactSwitch').simulate('change', false);
    expect(mockProps.onChange).toHaveBeenCalledWith(false);
  });
});
