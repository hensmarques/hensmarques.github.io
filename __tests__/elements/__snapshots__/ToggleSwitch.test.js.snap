// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch/> renders and matches the snapshot 1`] = `
<span
  className="toggle-container"
>
  <label>
    <ReactSwitch
      activeBoxShadow="0 0 2px 3px #3bf"
      boxShadow={null}
      checked={true}
      checkedIcon={
        <svg
          height="100%"
          style={
            Object {
              "position": "absolute",
              "top": 0,
            }
          }
          viewBox="-2 -5 17 21"
          width="100%"
        >
          <path
            d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      }
      className="switch"
      disabled={false}
      height={28}
      id="normal-switch"
      offColor="#888"
      offHandleColor="#fff"
      onChange={[MockFunction]}
      onColor="#7acd45"
      onHandleColor="#fff"
      uncheckedIcon={
        <svg
          height="100%"
          style={
            Object {
              "position": "absolute",
              "top": 0,
            }
          }
          viewBox="-2 -5 14 20"
          width="100%"
        >
          <path
            d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      }
      width={56}
    />
  </label>
</span>
`;
